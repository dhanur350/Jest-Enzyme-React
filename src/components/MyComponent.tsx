import { useState, useEffect } from "react"
const SecondComponent = () => {
    return <div>Hello</div>
}

const MyComponent = () => {
    const [click, setClick] = useState<boolean>(false);
    const [users, setUsers] = useState([])

    useEffect(() => {
        const fetchAllUsers = async () => {
            try {
                const response = await fetch('https://jsonplaceholder.typicode.com/users');
                const data = await response.json();
                setUsers(data);
            } catch (err) {
                console.log('Something went wrong!' + err);
            }
        }
        fetchAllUsers()
    }, [])
    function handleClick() {
        setClick(true);
    }
    console.log(users);

    return (
        <div>1
            <button onClick={handleClick}>Click here</button>
            {click && <div className="message">The button was clicked</div>}
            <SecondComponent />
            <>
                <h1>List of Users</h1>
                {users ? (
                    <ul>
                        {users.map(({ id, name }: any) => (
                            <li id={`li${id}`} key={id}>{name}</li>
                        ))}
                    </ul>
                ) : (
                    <p>No users found</p>
                )}
            </>
        </div>
    )
}

export default MyComponent