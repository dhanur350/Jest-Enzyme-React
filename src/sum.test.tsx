import { sum } from "./sum"
import { shallow } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import MyComponent from "./components/MyComponent";

Enzyme.configure({ adapter: new Adapter() });

let data = 12;
test("first test case", () => {
    expect(sum(5, 7)).toBe(data);
});


describe('<MyComponent />', () => {
    it('renders three components', () => {
        const wrapper = shallow(<MyComponent />);
        console.log(wrapper.debug);
        //   const children = wrapper.children();
        const childWrapper = wrapper.find("SecondComponent").shallow().find("div");
        expect(wrapper).toBeTruthy();
        expect(childWrapper.text()).toEqual("Hello");
    });
});

describe("<Button/> clicked or not", () => {
    it("Checking if button clicked or not", () => {
        const wrapper = shallow(<MyComponent />);
        expect(wrapper).toBeTruthy();
        expect(wrapper.find("div").text()).not.toContain("The button was clicked");
        wrapper.find("button").simulate("click");
        expect(wrapper.find(".message").text()).toContain("The button was clicked");
    })
})


describe('MyComponent', () => {
    it('should match snapshot', () => {
      const component = renderer.create(<MyComponent />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

// it("Second test", () => {
//     expect(data).toBe(12);
// })
