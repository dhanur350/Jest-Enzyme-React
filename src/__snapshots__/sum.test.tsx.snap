// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component Snapshot Checking if UI has changed or not 1`] = `
<div>
  1
  <button
    onClick={[Function]}
  >
    Click here
  </button>
  <div>
    Hello
  </div>
</div>
`;

exports[`MyComponent should match snapshot 1`] = `
<div>
  1
  <button
    onClick={[Function]}
  >
    Click here
  </button>
  <div>
    Hello
  </div>
</div>
`;
